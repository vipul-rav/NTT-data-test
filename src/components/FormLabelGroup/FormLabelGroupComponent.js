import React, { memo } from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Form from 'react-bootstrap/Form';

const FormLabelGroupComponent = memo(({ label, value }) => {
  return (
    <Row>
      <Form.Label column sm={2}>
        <storng>{label}</storng>
      </Form.Label>
      <Col>
        <Form.Label column sm={10}>
          {value}
        </Form.Label>
      </Col>
    </Row>
  );
});

export { FormLabelGroupComponent };
