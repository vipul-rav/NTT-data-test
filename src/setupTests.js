import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

const AllTheProviders = ({ children }) => {
  return <BrowserRouter>{children}</BrowserRouter>;
};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
