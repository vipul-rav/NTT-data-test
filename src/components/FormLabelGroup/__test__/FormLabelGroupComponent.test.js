import React from 'react';
import { cleanup } from '@testing-library/react';
import { render } from '../../../setupTests';
import { FormLabelGroupComponent } from '../FormLabelGroupComponent';

const setup = (overridesProps) => {
  const props = {
    label: 'test',
    value: 'test',
    ...overridesProps
  };

  const { container, getByText } = render(<FormLabelGroupComponent {...props} />);

  return {
    props,
    container,
    getByText
  };
};

describe('FormLabelGroupComponent', () => {
  afterEach(cleanup);
  it('should render', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
