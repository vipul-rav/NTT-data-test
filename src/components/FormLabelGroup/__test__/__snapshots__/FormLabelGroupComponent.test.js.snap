// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabelGroupComponent should render 1`] = `
<div>
  <div
    class="row"
  >
    <label
      class="form-label col-form-label col-sm-2"
    >
      <storng>
        test
      </storng>
    </label>
    <div
      class="col"
    >
      <label
        class="form-label col-form-label col-sm-10"
      >
        test
      </label>
    </div>
  </div>
</div>
`;
